pkg/tenant/api/v1/api.go

```diff
@@ -279,6 +281,25 @@ type ProjectTopicPage struct {
 	NextPageToken string   `json:"next_page_token,omitempty"`
 }
 
+type ProjectQueryRequest struct {
+	ProjectID string `json:"project_id"`
+	Query     string `json:"query"`
+}
+
+type ProjectQueryResponse struct {
+	Results []*QueryResult `json:"results"`
+	Error   string         `json:"error,omitempty"`
+}
+
+type QueryResult struct {
+	Metadata        map[string]string `json:"metadata"`
+	Mimetype        string            `json:"mimetype"`
+	Version         string            `json:"version"`
+	IsBase64Encoded bool              `json:"is_base64_encoded"`
+	Data            string            `json:"data"`
+	Created         string            `json:"created"`
```

Comment:
Let me know if I am missing any fields you are expecting here.

Reply:
In the Figma file [here](https://www.figma.com/file/PvL23fm6W36Zd88O1XVyIo/Beacon-DX-UI?type=design&node-id=8125-18524&mode=design&t=R8bRQrfWpjVBadm4-4), there's a # of results. Although, we'll only display 10, it appears that we want to show users the actual total even if it's more than 10. Should that be included here or handled elsewhere?

Reply:
@pdeziel at the top level of the response (e.g. outside the list of results), return the total number of events the query would return. For now, it will just be the length of the list; but we can update it in the future when we have an `Explain` command for Ensign.

Reply:
Thanks, makes sense to me.